From b67b7c15312bc3af4f63b7f39b74884268fab0bc Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:09 -0500
Subject: [drm] nv50/disp: fix dpms regression on certain boards

Message-id: <1418852380-13061-59-git-send-email-rclark@redhat.com>
Patchwork-id: 102190
O-Subject: [RHEL7 drm 58/89] drm/nv50/disp: fix dpms regression on certain boards
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: 3.17

commit 14f08b7c451164ac2b309a9ca615c5467fac6b7e

Author:     Ben Skeggs <bskeggs@redhat.com>
AuthorDate: Tue Sep 30 10:27:08 2014 +1000
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:45:33 2014 -0500

    drm/nv50/disp: fix dpms regression on certain boards

    commit 5838ae610ff36777b8fce6f353c2417980c1a1fa upstream

    Reported in fdo#82527 comment #2.

    Signed-off-by: Ben Skeggs <bskeggs@redhat.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/nouveau/core/engine/disp/nv50.c b/drivers/gpu/drm/nouveau/core/engine/disp/nv50.c
index 2283c44..90b0eb1 100644
--- a/drivers/gpu/drm/nouveau/core/engine/disp/nv50.c
+++ b/drivers/gpu/drm/nouveau/core/engine/disp/nv50.c
@@ -1570,9 +1570,10 @@ nv50_disp_intr_unk40_0_tmds(struct nv50_disp_priv *priv, struct dcb_output *outp
  const int   or = ffs(outp->or) - 1;
  const u32 loff = (or * 0x800) + (link * 0x80);
  const u16 mask = (outp->sorconf.link << 6) | outp->or;
+ struct dcb_output match;
  u8  ver, hdr;
 
- if (dcb_outp_match(bios, DCB_OUTPUT_DP, mask, &ver, &hdr, outp))
+ if (dcb_outp_match(bios, DCB_OUTPUT_DP, mask, &ver, &hdr, &match))
   nv_mask(priv, 0x61c10c + loff, 0x00000001, 0x00000000);
 }
 
-- 
1.7.1